components/style/gecko_string_cache/mod.rs

```diff
@@ -358,3 +340,21 @@ impl From<*mut nsIAtom> for Atom {
         }
     }
 }
+
+impl From<already_AddRefed<nsIAtom>> for Atom {
+    #[inline]
+    fn from(ptr: already_AddRefed<nsIAtom>) -> Atom {
+        let ptr = ptr.take();
+        debug_assert!(!ptr.is_null());
+        unsafe { Atom(WeakAtom::new(ptr)) }
+    }
+}
```

Comment:
This is unsound.

```rust
Atom::from(already_AddRefed::null())
```

Either make that `debug_assert` into `assert` or mark `already_AddRefed::null` as unsafe. I would prefer the latter.